[TASK] Update the contribution workflow documentation (#1282)
MyIntervals/emogrifier: .github/CONTRIBUTING.md

## General workflow

This is the workflow for contributing changes to this project::

> 1. [Fork the git repository](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project).

Author: Done.

2. Clone your forked repository locally and
   [install the development dependencies](#install-the-development-dependencies).
3. Add a local remote "upstream" so you will be able to

>    [synchronize your fork with the original repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork).

Reviewer: I know this doesn't relate specifically to this PR, but...

I think both 'remote' and 'upstream' should be in inverted-ticks, and I'm not sure 'local' belongs.  'A local remote 
"upstream"' does not appear to make any sense.  Though I don't fully understand what's going on here, and not sure I ever did that step.

The link given provides a link in turn for adding a `remote` `upstream", so an extra link for that probably isn't needed (though might be considered).

Author: I think I'll just remove that step. GitHub has added some UI for that anyway in the meantime.

## General workflow

This is the workflow for contributing changes to this project::

1. [Fork the Git repository](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project).
2. Clone your forked repository locally and
   [install the development dependencies](#install-the-development-dependencies).
3. Create a local branch for your changes.
4. [Add unit tests for your changes](#unit-test-your-changes).
   These tests should fail without your changes.
5. Add your changes. Your added unit tests now should pass, and no other tests
   should be broken. Check that your changes follow the same
   [coding style](#coding-style) as the rest of the project.
6. Add a changelog entry, newest on top.
7. [Commit](#git-commits) and push your changes.
8. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)
   for your changes.
9. Check that the CI build is green. (If it is not, fix the problems listed.)
   Please note that for first-time contributors, you will need to wait for a

>    maintainer to approve your pull request before the CI build will run.

Reviewer: Sorry to be nitpicky, but this implies that a maintainer would have approved the changes, whereas all they've done is clicked a button to allow the CI build to run.

```suggestion
   maintainer to allow your CI build to run.
```

   [coding style](#coding-style) as the rest of the project.
7. Add a changelog entry.
8. [Commit](#git-commits) and push your changes.
9. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)
   for your changes. Check that the CI build is green. (If it is not, fix the

>    problems listed.)

Reviewer: First-time contributors will need to wait for an admin to allow the CI build to run.  I think it's worth noting such here.
